City-of-Helsinki/berth-reservations-admin
# VEN-1190 | Change berth for existing lease

## Description :sparkles:

Change berth for existing lease

* `switchPlace` feature for berths
  * Uses `Offer` as base
  * Allowed usage of `Offer` without a boat, because leases don't currently have boats
* Wrapped `LeaseDetails` lease props inside `lease` object
* Made dropdowns inside modals overflow outside the modal body (by default)

## Issues :bug:

### Closes :no_good_woman:

* [VEN-1190](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1190): FE: change berth for a customer

## Testing :alembic:

### Automated tests :gear:️

* Added basic tests for `switchPlace` utils and `SwitchPlaceButton`
  * We should build a template for testing features that use queries and mutations
* Fixed warnings in `ApplicationDetails`, `BerthEditForm`, `ApplicationView`, `OpenInvoicesCard` and `unmarkedWsNoticeList` tests

### Manual testing :construction_worker_man:

* Go to a customer page with an active, paid berth
* Click "Vaihda venepaikka"
* Select a harbor and proceed
* Select a berth
* User should be returned to the customer page
* Previous lease should be terminated and a new lease with the new berth should be paid

## Screenshots 📸 

![image](https://user-images.githubusercontent.com/15479131/114028536-41f96a80-9881-11eb-90b4-968dc305a51c.png)
![image](https://user-images.githubusercontent.com/15479131/114028595-53427700-9881-11eb-9346-7779eef90bd4.png)
![image](https://user-images.githubusercontent.com/15479131/114028643-62292980-9881-11eb-8817-816af6318687.png)
![image](https://user-images.githubusercontent.com/15479131/114029010-c0eea300-9881-11eb-91e9-921404468cc7.png)
